storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1370,9 +1469,974 @@ public static int toPositive(int number) {
+                }
+            }
+            return null;
+        }
+    }
+
+    public static String localHostname () throws UnknownHostException {
+        return _instance.localHostnameImpl();
+    }
+
+    // Non-static impl methods exist for mocking purposes.
+    protected String localHostnameImpl () throws UnknownHostException {
+        return InetAddress.getLocalHost().getCanonicalHostName();
+    }
+
+    private static String memoizedLocalHostnameString = null;
+
+    public static String memoizedLocalHostname () throws UnknownHostException {
+        if (memoizedLocalHostnameString == null) {
+            memoizedLocalHostnameString = localHostname();
+        }
+        return memoizedLocalHostnameString;
+    }
+
+    /**
+     * Gets the storm.local.hostname value, or tries to figure out the local hostname
+     * if it is not set in the config.
+     * @param conf The storm config to read from
+     * @return a string representation of the hostname.
+    */
+    public static String hostname (Map<String, Object> conf) throws UnknownHostException  {
+        if (conf == null) {
+            return memoizedLocalHostname();
+        }
+        Object hostnameString = conf.get(Config.STORM_LOCAL_HOSTNAME);
+        if (hostnameString == null ) {
+            return memoizedLocalHostname();
+        }
+        if (hostnameString.equals("")) {
+            return memoizedLocalHostname();
+        }
+        return hostnameString.toString();
+    }
+
+    public static String uuid() {
+        return UUID.randomUUID().toString();
+    }
+
+    public static long secsToMillisLong(double secs) {
+        return (long) (1000 * secs);
+    }
+
+    public static Vector<String> tokenizePath (String path) {
+        String[] tokens = path.split("/");
+        Vector<String> outputs = new Vector<String>();
+        if (tokens == null || tokens.length == 0) {
+            return null;
+        }
+        for (String tok: tokens) {
+            if (!tok.isEmpty()) {
+                outputs.add(tok);
+            }
+        }
+        return outputs;
+    }
+
+    public static String parentPath(String path) {
+        if (path == null) {
+            return "/";
+        }
+        Vector<String> tokens = tokenizePath(path);
+        int length = tokens.size();
+        if (length == 0) {
+            return "/";
+        }
+        String output = "";
+        for (int i = 0; i < length - 1; i++) {  //length - 1 to mimic "butlast" from the old clojure code
+            output = output + "/" + tokens.get(i);
+        }
+        return output;
+    }
+
+    public static String toksToPath (Vector<String> toks) {
+        if (toks == null || toks.size() == 0) {
+            return "/";
+        }
+
+        String output = "";
+        for (int i = 0; i < toks.size(); i++) {
+            output = output + "/" + toks.get(i);
+        }
+        return output;
+    }
+    public static String normalizePath (String path) {
```

Comment:
There are versions of this and toksToPath already in Zookeeper.java we should just use those instead.